### ISSUE TITLE
Missing L-functions for EC over quadratic fields

### ISSUE THREAD

**Issue by edgarcosta** [LMFDB/lmfdb#2705]
At the moment there are 7586 isogeny classes of EC over quadratic fields without L-functions
7571 of which have base-field 2.0.7.1.
The remainder missing L-functions are listed below.
Some of them, due to their conductor were skipped on porpuse.
However, I don't understand why are we missing so many over 2.0.7.1, as I recall checking in the Summer of 2017 that we had everything that we wanted to have.

```
2.0.3.1-149769.3-CMa
2.0.3.1-149769.3-CMc
2.0.3.1-149769.3-CMb
2.0.218017403.1-1.0.1-a
2.0.11.1-42849.7-CMa
2.0.11.1-42849.9-CMa
2.0.3.1-149769.1-CMc
2.0.3.1-149769.1-CMb
2.0.3.1-149769.1-CMa
2.0.11.1-49729.3-CMa
2.0.451767203.1-1.0.1-a
2.0.443038171.1-1.0.1-a
2.0.349633643.1-1.0.1-a
2.0.11.1-49729.1-CMa
2.0.127285811.1-1.0.1-a
```

**Comment by JohnCremona** [comment]
The missing ones in the list above have 2 things in common: they have CM by the base field (which means that they are computed in a different way from the others since they do not have associated Bianchi newforms) and their conductors are large enough that they were added in the last batch of such curves which were later that year (committed in October 2017).

As to why there would be 7571 curves added late for that field: that is the exact number of isogeny classes (not curves) with conductor norm 40k-50k so it is very likely that this batch was the last to be uploaded, and done after the computation of the L-functions.

**Comment by edgarcosta** [comment]
That is indeed the case, as I finished these computations during the Summer of 2017.

**Comment by JohnCremona** [comment]
I have a few more elliptic curves to add soon, over imaginary quadratic fields.  (After this we will have all curves attached to Bianchi forms except those for which we can prove that there is no curve).  When I have done that I will you (@edgarcosta ) so you can add their L-functions.  Meanwhile we should make sure that the code is designed so that when there is a curve with no L-function stored, nothing breaks.

**Comment by edgarcosta** [comment]
#2706 takes care of avoiding broken links

**Comment by JohnCremona** [comment]
@edgarcosta  Can you update this?  #2851 lists 4 curves over 2.0.3.1, are the rest still missing also?

**Comment by edgarcosta** [comment]
I have not added a single ECNF L-function to the database since the issue has been opened, so everything stands as it is.

The ones over a quadratic field with a large discriminant will never be added, as the conductor of the L-fun will be too large.

**Comment by JohnCremona** [comment]
Fair enough for the last point.  Otherwise, we need a script to make it easy to compute and add L-functions as new curves are added, since otherwise this issue will recur every time curves are added.  One day we'll find and add the missing curves over degree 6 fields, for example.

**Comment by edgarcosta** [comment]
At some point (hopefully this summer) we will redo all the L-functions, and such a script will exist.

**Comment by AndrewVSutherland** [comment]
I've tagged this with the 1.2 milestone, which is largely focused on L-functions (see #3117).  The goal would be for 1.2 would be to compute all the L-functions we can readily compute and then close this issue (as @edgarcosta noted above, we are never going to be able to compute L-functions with extremely large conductor (given that we have to compute a_n up to roughly 16*sqrt(cond)), you can take 2^64 as an upper bound).

**Comment by JohnCremona** [comment]
No worries about the tiny number of curves with large conductor.  We are not planning to add more, and if we ever did they would continue to be exceptions.

**Comment by edgarcosta** [comment]
Also, there four more elliptic curves that are missing their L-function, copied from #2851
http://beta.lmfdb.org/EllipticCurve/2.0.3.1/113556.2/a/1
http://beta.lmfdb.org/EllipticCurve/2.0.3.1/116364.1/a/1
http://beta.lmfdb.org/EllipticCurve/2.0.3.1/146716.4/a/1
http://beta.lmfdb.org/EllipticCurve/2.0.3.1/146716.4/b/1
